platform/android/MapLibreAndroid/src/vulkan/java/org/maplibre/android/maps/renderer/surfaceview/MapLibreSurfaceView.java

```diff
@@ -0,0 +1,693 @@
+package org.maplibre.android.maps.renderer.surfaceview;
+
+import android.content.Context;
+import android.graphics.PixelFormat;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.view.SurfaceHolder;
+import android.view.SurfaceView;
+
+import androidx.annotation.NonNull;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class MapLibreSurfaceView extends SurfaceView implements SurfaceHolder.Callback2 {
```

Comment:
This class is similar enough tot he GL version to warrant using a common base class with some shared implementations.

Reply:
Similar to `TextureViewRenderThread`